agent/xds/jwt_authn_test.go

```diff
@@ -0,0 +1,153 @@
+	}
+
+	var (
+		oktaProvider = structs.IntentionJWTProvider{
+			Name: "okta",
+		}
+		auth0Provider = structs.IntentionJWTProvider{
+			Name: "auth0",
+		}
+		singleProviderIntention = &structs.IntentionJWTRequirement{
+			Providers: []*structs.IntentionJWTProvider{
+				&oktaProvider,
+			},
+		}
+		multiProviderIntentions = &structs.IntentionJWTRequirement{
+			Providers: []*structs.IntentionJWTProvider{
+				&oktaProvider,
+				&auth0Provider,
+			},
+		}
+		remoteCE = map[string]*structs.JWTProviderConfigEntry{
+			"okta": {
+				Kind:   "jwt-provider",
+				Name:   "okta",
+				Issuer: "test-issuer",
+				JSONWebKeySet: &structs.JSONWebKeySet{
+					Remote: &structs.RemoteJWKS{
+						FetchAsynchronously: true,
+						URI:                 "https://example.com/.well-known/jwks.json",
+					},
+				},
+			},
+			"auth0": {
+				Kind:   "jwt-provider",
+				Name:   "auth0",
+				Issuer: "another-issuer",
+				JSONWebKeySet: &structs.JSONWebKeySet{
+					Remote: &structs.RemoteJWKS{
+						FetchAsynchronously: true,
+						URI:                 "https://example.com/.well-known/jwks.json",
+					},
+				},
+			},
+		}
+		localCE = map[string]*structs.JWTProviderConfigEntry{
+			"okta": {
+				Kind:   "jwt-provider",
+				Name:   "okta",
+				Issuer: "test-issuer",
+				JSONWebKeySet: &structs.JSONWebKeySet{
+					Local: &structs.LocalJWKS{
+						JWKS: "eyJrZXlzIjogW3sKICAiY3J2IjogIlAtMjU2IiwKICAia2V5X29wcyI6IFsKICAgICJ2ZXJpZnkiCiAgXSwKICAia3R5IjogIkVDIiwKICAieCI6ICJXYzl1WnVQYUI3S2gyRk1jOXd0SmpSZThYRDR5VDJBWU5BQWtyWWJWanV3IiwKICAieSI6ICI2OGhSVEppSk5Pd3RyaDRFb1BYZVZuUnVIN2hpU0RKX2xtYmJqZkRmV3EwIiwKICAiYWxnIjogIkVTMjU2IiwKICAidXNlIjogInNpZyIsCiAgImtpZCI6ICJhYzFlOGY5MGVkZGY2MWM0MjljNjFjYTA1YjRmMmUwNyIKfV19",
+					},
+				},
+			},
+		}
+		pWithOneProvider = &structs.IntentionPermission{
+			Action: structs.IntentionActionAllow,
+			HTTP: &structs.IntentionHTTPPermission{
+				PathPrefix: "/some-special-path",
+			},
+			JWT: singleProviderIntention,
+		}
+		permWithPath = &structs.IntentionPermission{
+			Action: structs.IntentionActionAllow,
+			HTTP: &structs.IntentionHTTPPermission{
+				PathPrefix: "/some-special-path",
+			},
+			JWT: singleProviderIntention,
+		}
+	)
+
+	tests := map[string]struct {
+		intentions structs.SimplifiedIntentions
+		provider   map[string]*structs.JWTProviderConfigEntry
+	}{
+		"remote-provider": {
+			intentions: simplified(testIntention(t, ixnOpts{src: "web", action: structs.IntentionActionAllow, jwt: singleProviderIntention})),
+			provider:   remoteCE,
+		},
+		"local-provider": {
+			intentions: simplified(testIntention(t, ixnOpts{src: "web", action: structs.IntentionActionAllow, jwt: singleProviderIntention})),
+			provider:   localCE,
+		},
+		"intention-with-path": {
+			intentions: simplified(testIntention(t, ixnOpts{src: "web", action: structs.IntentionActionAllow, perms: pWithOneProvider})),
+			provider:   remoteCE,
+		},
+		"top-level-provider-with-permission": {
```

Comment:
The JWT config can be specified [in a top-level field of an intention](https://github.com/hashicorp/consul/blob/d20e3df63c0a2dbb13b67e0d7b4023f8c8b91da5/agent/structs/config_entry_intentions.go#L24), in which case it applies to all requests, or [within the intention permissions](https://github.com/hashicorp/consul/blob/d20e3df63c0a2dbb13b67e0d7b4023f8c8b91da5/agent/structs/config_entry_intentions.go#L370) where it can apply based on an HTTP request match